### PR TITLE
Only catch OOM if `grad_accum='auto'`

### PR DESCRIPTION
**Pull request by hanlint**
This PR fixes a reported bug where gradient accumulation was always automatic.

Coming along for the ride -> some better logging of the grad accum increases.

### CODE REVIEW

**Review by moinnadeem — APPROVED** [review]
Can confirm that I am seeing OOMs again

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -1645,7 +1645,7 @@ def _train_batch(self, use_grad_scaling: bool):
                         else:
                             optimizer.step()
             except RuntimeError as e:
-                if _is_cuda_oom(e):
+                if self.adaptive_gradient_accumulation and _is_cuda_oom(e):
```
Comment (hanlint) [inline composer/trainer/trainer.py:1648]: @ravi-mosaicml is it OK to do the check here? or should we always run the `should_handle_cuda_oom` logic, and then reraise it so that all the ranks fail?

**Comment by hanlint** [comment]
Merging now -- created CO-535 to track test cases for automatic gradient accumulation.
